aten/src/ATen/native/quantized/cpu/fake_quantize_per_tensor_affine.cpp

```diff
@@ -21,49 +22,26 @@ namespace {
 Notes:
   - quant_delay might be set to non-zero to help weights stabilize in the
     beginning of the training.
-  - quantization range [0, 2^bits - 1]
+  - quantization range [quant_min, quant_max]
 */
-
-class FakeQuantizePerTensorAffineOp_forward : public c10::OperatorKernel {
- public:
-  at::Tensor operator()(
-      at::Tensor X,
+Tensor fake_quantize_per_tensor_affine_cpu(
+      const Tensor& self,
       double scale,
       int64_t zero_point,
-      int64_t quant_min = 0,
-      int64_t quant_max = 255,
-      int64_t quant_delay = 0,
-      int64_t iter = 0
+      int64_t quant_min,
+      int64_t quant_max
     ) {
-    // Sanity checks.
-    if (quant_min > quant_max) {
-      throw std::invalid_argument("`quant_min` should be less than or equal to `quant_max`.");
-    }
-    if (zero_point < 0) {
-      throw std::invalid_argument("`zero_point` must be a positive integer.");
-    }
-    if (quant_delay < 0) {
-      throw std::invalid_argument("`quant_delay` must be a positive integer.");
-    }
-
-    if (quant_delay != 0 && iter < 0) {
-      throw std::invalid_argument(
-        "`iter` must be >=0 for non-zero `quant_delay`");
-    }
+    TORCH_CHECK(self.scalar_type() == ScalarType::Float);
+    TORCH_CHECK(quant_min <= quant_max, "`quant_min` should be less than or equal to `quant_max`.");
+    TORCH_CHECK(zero_point >= 0, "`zero_point` must be a positive integer.");
 
-    auto Y = at::empty_like(X);
-
-    if (quant_delay > 0 && iter <= quant_delay) {
-      Y.copy_(X);  // We might want to just return the input here.
-      return Y;
-    }
+    auto Y = at::empty_like(self);
 
     double inv_scale = 1.0f / scale;
-    Y = (((X * inv_scale + 0.5f).floor() + zero_point)
+    Y = (((self * inv_scale + 0.5f).floor() + zero_point)
```

Comment:
To be consistent with the quantizer implementation and the GPU implementation, let us floor after adding zero-point.  (cf: https://github.com/pytorch/pytorch/blob/master/test/common_utils.py#L1086)